zerver/webhooks/beeminder/view.py

```diff
@@ -0,0 +1,57 @@
+# Webhooks for external integrations.
+from typing import Text
+from django.http import HttpRequest, HttpResponse
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.actions import check_send_stream_message, check_send_private_message
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile, get_user_profile_by_email
+import time
+current_time = time.time()
+
+@api_key_only_webhook_view("beeminder")
+@has_request_variables
+def api_beeminder_webhook(request, user_profile: UserProfile,
+                          payload=REQ(argument_type='body'),
+                          stream=REQ(default="beeminder"),
+                          email=REQ(default='foo@gmail.com'),
+                          topic=REQ(default='beekeeper')):
+
+    # type: (HttpRequest, UserProfile, Dict[str, Any], str, email, str) -> HttpResponse
+    secret = payload["goal"]["secret"]
+    goal_name = payload["goal"]["slug"]
+    losedate = payload["goal"]["losedate"]
+    limsum = payload["goal"]["limsum"]
+    pledge = payload["goal"]["pledge"]
+    time_remain = (losedate - current_time)/3600   # time in hours
+    # To show user's probable reaction by looking at pledge amount
+    if pledge > 0:
+        expression = ':worried:'
+    else:
+        expression = ':relieved:'
+
+    if not secret:
+        # In this case notifications will be sent to stream
+        def get_user_name(email):
+            p = get_user_profile_by_email(email)   # email here is user's zulip mail id
+            if p.email == email:
+                n = p.short_name
+            else:
+                n = "%s".format(email)
+            return n
```

Comment:
Okay, so a couple of points:
* I don't think you need the comment about `# email here is user's zulip mail id`.
* Move the function `get_user_name` out of the main webhook function.
* Also, I would recommend using variable names that are more descriptive than `p` and `n`. It makes the code more readable. :)